File: BubbleSort.go
package main

import (
	"fmt"
	"math/rand"
	"time"
)

func bubblesort(a []int) {
	for itemCount := len(a) - 1; ; itemCount-- {
		hasChanged := false
		for index := 0; index < itemCount; index++ {
			if a[index] > a[index+1] {
				a[index], a[index+1] = a[index+1], a[index]
				hasChanged = true
			}
		}
		if hasChanged == false {
			break
		}
	}
}

func main() {
	rand.Seed(time.Now().Unix())
	list := rand.Perm(10)
	fmt.Println("Unsorted array:", list)

	bubblesort(list)
	fmt.Println("Sorted array:", list)
}

File: BubbleSort.jl
function bubblesort!(arr::AbstractVector)
    for _ in 2:length(arr), j in 1:length(arr)-1
        if arr[j] > arr[j+1]
            arr[j], arr[j+1] = arr[j+1], arr[j]
        end
    end
    return arr
end
 
array = rand(1:100, 5)
println("Unsorted array: $array\nSorted array: ", bubblesort!(array))